gax-java/gax-grpc/src/main/java/com/google/api/gax/grpc/GrpcCallContext.java

```diff
@@ -602,7 +638,9 @@ public GrpcCallContext withChannel(Channel newChannel) {
         options,
         retrySettings,
         retryableCodes,
-        endpointContext);
+        endpointContext,
+        // Defaults to false again since we cannot tell whether the channel is DirectPath.
+        false);
```

Comment:
This could lead to a potential bug. If `GrpcCallContext` contains a direct path channel, and it is re-created with another direct path channel using this method, then we would send duplicated headers again. 
Fortunately I don't see any usage of this method at this moment, but with the introduction of `isDirectPath` in `GrpcCallContext`, we should probably mark this method as `@ObsoleteApi`, and suggest everyone to use `withTransportChannel` instead. See an example [here](https://github.com/googleapis/sdk-platform-java/blob/3857e3f6dc633a9adaf29e9770a584419e2f2214/gax-java/gax/src/main/java/com/google/api/gax/rpc/ApiCallContext.java#L70-L71).

Reply:
Thanks. I checked that this API is only used in two test classes. So it should be safe to be marked as obsolete. Followed your example.